driver-core/src/main/com/mongodb/client/model/expressions/StringExpression.java

```diff
@@ -25,34 +25,142 @@
+     *
+     * @param start the start index in UTF-8 code points.
+     * @param length the length in UTF-8 code points.
+     * @return the resulting value.
+     */
     default StringExpression substr(final int start, final int length) {
         return this.substr(of(start), of(length));
     }
 
+    /**
+     * The substring of {@code this} string, from the {@code start} index
+     * inclusive, and including the specified {@code length}, up to
+     * the end of the string.
+     *
+     * <p>The index position is in UTF-8 encoded bytes, not in
+     * UTF-8 code points.
+     *
+     * @param start the start index in UTF-8 encoded bytes.
+     * @param length the length in UTF-8 encoded bytes.
+     * @return the resulting value.
+     */
     StringExpression substrBytes(IntegerExpression start, IntegerExpression length);
 
+    /**
+     * The substring of {@code this} string, from the {@code start} index
+     * inclusive, and including the specified {@code length}, up to
+     * the end of the string.
+     *
+     * <p>The index position is in UTF-8 encoded bytes, not in
+     * UTF-8 code points.
+     *
+     * @param start the start index in UTF-8 encoded bytes.
+     * @param length the length in UTF-8 encoded bytes.
+     * @return the resulting value.
+     */
     default StringExpression substrBytes(final int start, final int length) {
         return this.substrBytes(of(start), of(length));
     }
 
+    /**
+     * Converts {@code this} string to an {@linkplain IntegerExpression integer}.
+     *
+     * <p>This will cause an error if this string does not represent an integer.
+     *
+     * @return the resulting value.
+     */
     IntegerExpression parseInteger();
 
+    /**
+     * Converts {@code this} string to a {@linkplain DateExpression date}.
+     *
+     * <p>This will cause an error if this string does not represent a valid
+     * date string (such as "2018-03-20", "2018-03-20T12:00:00Z", or
+     * "2018-03-20T12:00:00+0500").
+     *
+     * @return the resulting value.
+     */
     DateExpression parseDate();
 
+    /**
+     * Converts {@code this} string to a {@linkplain DateExpression date},
+     * using the specified {@code format}.
+     *
+     * @mongodb.driver.manual reference/operator/aggregation/dateToString/#std-label-format-specifiers Format Specifiers
+     * @param format the format.
+     * @return the resulting value.
+     */
     DateExpression parseDate(StringExpression format);
 
+    /**
+     * Converts {@code this} string to a {@linkplain DateExpression date},
+     * using the specified {@code timezone} and {@code format}.
+     *
+     * @mongodb.driver.manual reference/operator/aggregation/dateToString/#std-label-format-specifiers Format Specifiers
+     * @param format the format.
+     * @param timezone the UTC Offset or Olson Timezone Identifier.
+     * @return the resulting value.
+     */
     DateExpression parseDate(StringExpression timezone, StringExpression format);
```

Comment:
We should document that this method errors if the parsed data (and, correspondingly, the format) contains timezone information.